module/VuFind/src/VuFind/Db/Service/PagesService.php

```diff
@@ -0,0 +1,163 @@
+    {
+    }
+
+    /**
+     * Insert a new page into the database or update an existing one.
+     *
+     * @param array $data     Data to be written to the database
+     * @param array $pageData Data of an existing page
+     * @param int   $page_id  Id of the page to be edited
+     *
+     * @throws Exception
+     */
+    public function insertOrUpdatePage($data, $pageData = null, $page_id = null): void
+    {
+        $this->pages->insertOrUpdatePage($data, $pageData, $page_id);
+    }
+
+    /**
+     * Get a list of pages from the database
+     *
+     * @param array $where Filter setting for the request
+     * @param array $order Order settings for the request
+     */
+    public function getPagesList($where = null, $order = null): array
+    {
+        return $this->pages->getPagesList($where, $order);
+    }
+
+    /**
+     * Get all data for a page
+     *
+     * @param array $page_id Id of the page
+     */
+    public function getPagesDataByPageId($page_id): array
+    {
+        return $this->pages->getPagesDataByPageId($page_id);
+    }
+
+    /**
+     * Get a page object by id
+     *
+     * @param int $id Id of the page
+     *
+     * @return mixed page object
+     */
+    public function getPageById($id): ?PagesEntityInterface
+    {
+        return $this->pages->getPageById($id);
+    }
+
+    /**
+     * Get all page objects with the same page_id
+     *
+     * @param int $page_id Id of the page
+     *
+     * @return mixed Array of page objects
+     */
+    public function getPagesByPageId($page_id): mixed
+    {
+        return $this->pages->getPagesByPageId($page_id);
+    }
+
+    /**
+     * Get a page object by page_id and language
+     *
+     * @param int    $page_id  Id of the page
+     * @param string $language Language of the page
+     *
+     * @return mixed page object
+     */
+    public function getPageByPageIdAndLanguage($page_id, $language): ?PagesEntityInterface
+    {
+        return $this->pages->getPageByPageIdAndLanguage($page_id, $language);
+    }
+
+    /**
+     * Set the priority of a page
+     *
+     * @param int $index   New position of the page
+     * @param int $page_id Id of the page
+     */
+    public function setPriorityForPageId($index, $page_id): void
+    {
+        $this->pages->setPriorityForPageId($index, $page_id);
+    }
+
+    /**
+     * Set the visibility of a page
+     *
+     * @param int $visibility New visibility of the page
+     * @param int $page_id    Id of the page
+     */
+    public function setVisibilityForPageId($visibility, $page_id): void
+    {
+        $this->pages->setVisibilityForPageId($visibility, $page_id);
+    }
+
+    /**
+     * Set the global visibility of a page
+     *
+     * @param int $visibility_global New visibility of the page
+     * @param int $page_id           Id of the page
+     */
+    public function setVisibilityGlobalForPageId($visibility_global, $page_id): void
```

Comment:
Should this be `setGlobalVisibilityForPageId` rather than `setVisibilityGlobalForPageId`?

Reply:
I ws following the name of the variables (`visibility` and `visibiliy_global`) throughout the code.

Reply:
Ahh, I see, that makes sense!